### PR TITLE
Support query params

### PR DESCRIPTION
**Pull request by mhkeller**
In working on https://github.com/sindresorhus/electron-serve/pull/36, I also wanted the ability to pass in query parameters to different files. 

I added an example in the README since I didn't see any explicit documentation for the arguments for the returned function. Again, feel free to make any fixes or adjustments.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -72,7 +72,8 @@ module.exports = options => {
 		session.protocol.registerFileProtocol(options.scheme, handler);
 	});
 
-	return async window_ => {
-		await window_.loadURL(`${options.scheme}://${options.hostname}`);
+	return async (window_, query) => {
+		const queryString = query ? `?${query}` : '';
```
Comment (sindresorhus) [inline index.js]: This is not safe. Would be better to accept a `searchParams` object that is passed `new URLSearchParams`.

This is also missing docs, types, and test.

Comment (mhkeller) [reply index.js]: Sorry I'm not completely following. Do you mean something like this:

```js
return async (window_, searchParams) => {
  const qs = searchParams ? '?' + new URLSearchParams(searchParams).toString() : ''
  await window_.loadURL(`${options.scheme}://${options.hostname}${qs}`);
};
```

Comment (sindresorhus) [reply index.js]: Yes 👍

File: readme.md [hunk 2]
```diff
@@ -79,6 +81,23 @@ Default: [`electron.session.defaultSession`](https://electronjs.org/docs/api/ses
 
 The [partition](https://electronjs.org/docs/api/session#sessionfrompartitionpartition-options) the protocol should be installed to, if you're not using Electron's default partition.
```
Comment (sindresorhus) [inline readme.md:91]: Line 42:

```
### loadUrl = serve(options)

File: readme.md [hunk 3]
```diff
@@ -79,6 +81,23 @@ Default: [`electron.session.defaultSession`](https://electronjs.org/docs/api/ses
...
 
+### loadUrl(window, searchParams?)
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
### loadUrl(window, searchParameters?)
```

File: index.d.ts [hunk 4]
```diff
@@ -41,7 +41,7 @@ declare namespace electronServe {
 	/**
 	Load the index file in the window.
 	*/
-	type loadURL = (window: BrowserWindow) => Promise<void>;
+	type loadURL = (window: BrowserWindow, searchParameters?: Record<string>) => Promise<void>;
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
	type loadURL = (window: BrowserWindow, searchParameters?: Record<string, string> | URLSearchParams) => Promise<void>;
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: readme.md (hunk 2 above)
Comment (mhkeller) [reply readme.md:91]: What's the change needed here? The `searchParameters` option goes on the returned function which I added in a new section at line 91. Unless I'm missing something...

**Comment by mhkeller** [comment]
Thanks for the help and attention on this!

**Pull request merged**